platforms/common/include/px4_platform_common/sem.h

```diff
@@ -48,6 +48,9 @@
 #define SEM_PRIO_INHERIT          1
 #define SEM_PRIO_PROTECT          2
 #define sem_setprotocol(s,p)
+#define px4_mutex_init(m)         px4_sem_init(m, 0, 1)
```

Comment:
Perhaps we should keep the shared parameter? It is not used in NuttX, but I  am not sure about linux